src/components/Table/common/utils.ts

```diff
@@ -164,31 +147,27 @@ export function getEditColumnOptions<T extends RowData>(
   const { getAllColumns, initialState } = table;
   const { columnVisibility: initialVisibilityState } = initialState;
   const allColumns = getAllColumns();
-  return allColumns.reduce(
-    (
-      acc,
-      {
-        columnDef: { header },
-        getCanHide,
-        getIsVisible,
-        getToggleVisibilityHandler,
-        id,
-      }
-    ) => {
-      if (getCanHide()) {
-        const option: CheckboxMenuListItem = {
-          checked: getIsVisible(),
-          disabled: initialVisibilityState[id],
-          label: header as string, // TODO revisit type assertion here
-          onChange: getToggleVisibilityHandler(),
-          value: id,
-        };
-        acc.push(option);
-      }
-      return acc;
-    },
-    [] as CheckboxMenuListItem[]
-  );
+  return allColumns.reduce((acc, column) => {
+    const {
+      columnDef: { header },
+      getCanHide,
+      getIsVisible,
+      id,
+    } = column;
+    if (getCanHide()) {
+      const option: CheckboxMenuListItem = {
+        checked: getIsVisible(),
+        disabled: initialVisibilityState[id],
+        label: header as string, // TODO revisit type assertion here
+        onChange: () => {
+          handleToggleVisibility(table, column);
```

Comment:
Updated the `onChange` function to `handleToggleVisibility`.

Reply:
Note: The `getEditColumnOptions` function will be updated as part of the fix for column visibility.